### PR TITLE
console_conf: various recover chooser tweaks

### PR DESCRIPTION
**Pull request by bboozzoo**
A bunch of tweaks for the recovery chooser UI based on the discussion with @niemeyer.

Most important changes:
- drop the ABORT button
- introduce BACK button, goes back to the correct previous screen
- sort actions in the order of run -> recover -> install -> rest
- capitalize action title strings
- simplify current model actions screen
- user hints in the confirm screen for well known modes
- generic hint for unknown modes
- use model's display name in the top banner
- use `(installed)` instead of `(current)` in the all-systems screen for the current system

### CODE REVIEW

**Comment by bboozzoo** [comment]
@xnox please don't merge it yet. I'll let you know when it's ready.

File: console_conf/ui/views/chooser.py [hunk 1]
```diff
@@ -144,56 +151,67 @@ def __init__(self, controller, systems):
...
+        self.controller.back()
 
 
-class ChooserConfirmView(BaseView):
-    title = "Ubuntu Core"
-    excerpt = ("Summary of the selected action.")
+class ChooserConfirmView(ChooserBaseView):
+
+    canned_summary = {
+        "run": "Continue running the system without any changes.",
```
Comment (bboozzoo) [inline console_conf/ui/views/chooser.py:176]: @degville would appreciate your input on the messages here. This is presented to the user as a summary of the selected action.

The `default_summary` is a fallback messaage we use for an action that we don't know about.

Comment (degville) [reply console_conf/ui/views/chooser.py:176]: As I know you want to merge this, I've had a quick glance over the text and it looks good.

The only thing I'd probably tweak is the default summary to something like 'Make sure you understand the consequences of performing this action.`, or maybe add a little more help (or a link) on what those consequences might be.

File: console_conf/ui/views/welcome.py [hunk 2]
```diff
@@ -51,6 +51,13 @@ def confirm(self, result):
...
+        "System mode selection triggered. Proceed to select one of available "
+        "systems and actions."
     )
+
+    def __init__(self, controller):
+        current = controller.model.current
+        if current is not None:
+            self.title = current.model.display_name
+
+        super().__init__(controller)
```
Comment (zyga) [inline console_conf/ui/views/welcome.py:63]: Could this init happen before we assign to `self.title`?

File: console_conf/ui/views/chooser.py [hunk 3]
```diff
@@ -47,23 +47,36 @@
...
+        if current is not None:
+            self.title = current.model.display_name
+
+        super().__init__(scr)
+
+
+def by_preferred_action_type(action):
+    """Order action entries by having the 'run' mode first, then 'recover', then
+    'install', the rest is ordered alphabetically."""
+    return (action.mode != "run",
```
Comment (zyga) [inline console_conf/ui/views/chooser.py:63]: I find this key function unusual as it returns a tuple (bool, bool, bool, str), I was expecting a (int, str)

```
priority = {"run": 0, "recover": 1, "install": 2}
return (priority.get(action.mode, 100), action.title.lower())
```

File: console_conf/ui/views/chooser.py [hunk 4]
```diff
@@ -47,23 +47,36 @@
...
     title = "Ubuntu Core"
 
-    def __init__(self, controller, current, has_more=False):
-        fmt = "Select one of available actions for \"{}\" by \"{}\"{}."
-        maybe_more = " or view all available systems" if has_more else ""
-        excerpt = fmt.format(current.model.display_name,
-                             current.brand.display_name,
-                             maybe_more)
+    def __init__(self, current, scr):
+        if current is not None:
```
Comment (zyga) [inline console_conf/ui/views/chooser.py:54]: Same question, could we call `__init__()` as the first thing and only then assign to `self.title`? I find this style more in line with what ought to happen in a perfect scenario.

File: console_conf/ui/views/chooser.py (hunk 3 above)
Comment (bboozzoo) [reply console_conf/ui/views/chooser.py:63]: Nice! I'll push that

File: console_conf/ui/views/welcome.py (hunk 2 above)
Comment (bboozzoo) [reply console_conf/ui/views/welcome.py:63]: Fixed this and the other one.

**Comment by bboozzoo** [comment]
@xnox this is ready now.

**Comment by bboozzoo** [comment]
Thanks!

**Pull request merged**